src/MOI/callbacks/MOI_heuristic.jl

```diff
@@ -0,0 +1,88 @@
+function moi_heuristic_xprs_optnode_wrapper(func::Function, model::Optimizer, callback_data::CD) where {CD<:CallbackData}
+    info = model.callback_table.moi_heuristic::Union{Tuple{CallbackInfo{CD}}, Nothing}
+
+    if !isnothing(info)
+        push_callback_state!(model, CS_MOI_HEURISTIC)
```

Comment:
This is not thread-safe, right?
We should probably comment it

Reply:
Certainly not.

After reading the [docs](https://www.fico.com/fico-xpress-optimization/docs/dms2019-04/solver/optimizer/HTML/chapter5_sec_section4004.html), I think that we shouldn't worry about that much since it only breaks if [`MUTEXCALLBACKS`](https://www.fico.com/fico-xpress-optimization/docs/dms2019-04/solver/optimizer/HTML/MUTEXCALLBACKS.html) is set to `0`. We could make this mechanism thread-safe when the mutex is off but it would take some time thinking to get things right.

Reply:
Lets add this info somewhere?
Maybe in the optimizer struct right by the associated fields

Reply:
How about triggering a warning if the number of threads is greater than one and there are registered callbacks?  @joaquimg